### DISCUSSION TITLE
How to remove bookmarks/outlines?

### DISCUSSION THREAD

**Discussion by mitchpaulus** [py-pdf/pypdf#1427] [Q&A]
I'm looking to remove existing bookmarks/outlines from a PDF when merging. There is the `add_outline_item` but I did not see any methods corresponding to removal. A simple web search didn't turn up anything useful. 

Is there a canonical way to do this within PyPDF2?

**Comment by pubpub-zz** [comment]
the append/merge functions has a parameters to not copy the outlines. to remove a single outline, there is currently no function to easily remove it. Will prepare something

**Reply by brunofauth** [reply]
Hi, is there any workaround for that as of yet?

**Reply by stefan6419846** [reply]
As far as I understand the linked issue, this should work (although I have not tested it):

```
from pypdf import PdfWriter

writer = PdfWriter(clone_from='file.pdf')
for outline in writer.outline:
    outline.remove_from_tree()
writer.write('out.pdf')
```

If it works for you, consider opening a PR to document this.

**Comment by BH2WFR** [comment]
I also don't know how to remove all bookmarks...

**Reply by stefan6419846** [reply]
Did you try https://github.com/py-pdf/pypdf/discussions/1427#discussioncomment-12825035?

**Reply by BH2WFR** [reply]
> Did you try [#1427 (reply in thread)](https://github.com/py-pdf/pypdf/discussions/1427#discussioncomment-12825035)?

Yes, but it throws an exception:
```
Traceback (most recent call last):
File "d:\test\add-pdf-bookmarks.py", line 131, in <module>
outline.remove_from_tree()
~~~~~~~~~~~~~~~~~~~~~~~~^^
File "C:\Program_Files\anaconda3\Lib\site-packages\pypdf\generic_data_structures.py", line 870, in remove_from_tree
raise ValueError("Removed child does not appear to be a tree item")
ValueError: Removed child does not appear to be a tree item
```

My Code:
```python
    reader = pypdf.PdfReader(filepath)
    writer = pypdf.PdfWriter(clone_from=reader)
    for outline in writer.outline:
        outline.remove_from_tree()
```

**Reply by stefan6419846** [reply]
`generic_data_structures.py` looks somehow odd. Please provide a reproducing example.
